x-pack/apm-server/sampling/eventstorage/partition_rw.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package eventstorage
+
+import (
+	"errors"
+
+	"github.com/elastic/apm-data/model/modelpb"
+)
+
+// PartitionReadWriter reads from and writes to storage across partitions.
+type PartitionReadWriter struct {
+	s *Storage
+}
+
+// Flush waits for preceding writes to be committed to storage.
+func (rw *PartitionReadWriter) Flush() error {
+	return nil
```

Comment:
Why is this a no-op? Perhaps we should have another interface separate from RW that has Flush? Maybe:
 - Reader
 - Writer
 - Flusher
 - ReadWriter
 - ReadWriteFlusher

(Perhaps a _little_ over the top, but avoids having methods just because we're implementing an overload wide interface)

Reply:
> Why is this a no-op?

All `Flush` are no-op at the moment, because we stopped using transaction/"batch". In the underlying implementation, all writes are committed immediately using a temporary batch. This has a side effect that if a read follows very closely to a concurrent write, and the read returned sooner than the write, I _think_ they may race and the read may return NotFound. We'll need to add back sharding and enforce strict ordering for trace ID concurrent RW if we need to fix this edge case.

> Perhaps we should have another interface separate from RW that has Flush?

I should have cleaned this up. RW Flush (i.e. transaction commit) is not used, while storage manager flush (i.e. DB flush to disk) is used on apm-server exit.